### ISSUE TITLE
assert(to_scroll >= 0) in msg_scroll_flush sometimes fails (especially with noice.nvim)

### ISSUE THREAD

**Issue by ddickstein** [neovim/neovim#21056]
### Neovim version (nvim -v)

0.7.0

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

CentOS 7.9

### Terminal name/version

xfce4-terminal 0.8.7.4

### $TERM environment variable

tmux-256color

### Installation

system package manager

### How to reproduce the issue

[This assertion](https://github.com/neovim/neovim/blob/release-0.7/src/nvim/message.c#L2360) sometimes fails. This should be able to reproduce it:
```
$ nvim --headless -u NONE -i /tmp/shada --cmd 'set shada+=%' -c q /tmp/crash.txt
$ false | nvim --headless --clean -i /tmp/shada --cmd 'set shada+=%' -c scriptnames -c q -
```

### Expected behavior

Commands run without any problems

### Actual behavior

The second command crashes with the following:
```
E575: Error while reading ShaDa file: buffer list at position 79 contains entry with invalid line number
  1: /usr/share/nvim/runtime/ftplugin.vim
  2: /usr/share/nvim/runtime/indent.vim
  3: /usr/share/nvim/runtime/filetype.lua
  4: /usr/share/nvim/runtime/filetype.vim
  5: /usr/share/nvim/runtime/syntax/syntax.vim
  6: /usr/share/nvim/runtime/syntax/synload.vim
  7: /usr/share/nvim/runtime/plugin/gzip.vim
  8: /usr/share/nvim/runtime/plugin/health.vim
  9: /usr/share/nvim/runtime/plugin/man.vim
 10: /usr/share/nvim/runtime/plugin/matchit.vim
 11: /usr/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
 12: /usr/share/nvim/runtime/plugin/matchparen.vim
 13: /usr/share/nvim/runtime/plugin/netrwPlugin.vim
 14: /usr/share/nvim/runtime/plugin/rplugin.vim
 15: /usr/share/nvim/runtime/plugin/shada.vim
 16: /usr/share/nvim/runtime/plugin/spellfile.vim
 17: /usr/share/nvim/runtime/plugin/tarPlugin.vim
 18: /usr/share/nvim/runtime/plugin/tohtml.vim
 19: /usr/share/nvim/runtime/plugin/tutor.vim
 20: /usr/share/nvim/runtime/plugin/zipPlugin.vim
 21: /usr/share/nvim/runtime/scripts.vimnvim: /path/to/src/nvim/message.c:2360: msg_scroll_flush: Assertion `to_scroll >= 0' failed.
Aborted (core dumped)
```

**Comment by zeertzjq** [comment]
v0.7.0 is too old. Can you try nightly appimage?

**Comment by zeertzjq** [comment]
Does the appimage also not work on the machine?

**Comment by ddickstein** [comment]
Didn't test with appimage, but did test on my personal machine which runs 0.8.2 and was able to reproduce the problem as described, so I don't think it's been fixed since 0.7.0.

**Comment by zeertzjq** [comment]
Did you test nightly version?

**Comment by ddickstein** [comment]
No, but building nightly can't be a requirement for filing bug reports.

**Comment by zeertzjq** [comment]
Can't you use the appimage?

**Comment by zeertzjq** [comment]
Also try v0.8.3 when it is available

**Comment by ddickstein** [comment]
Tested on the nightly appimage - v0.9.0-dev-867+gf9826e1df - problem is still present.

**Comment by TechnicalDC** [comment]
I have also tested on the nightly - v0.10.0-dev-516+ga7e5d4238 - problem is still present. It happens randomly.

**Comment by arpangreat** [comment]
It's still happening on my side (version: v0.10.0-dev-567+g4dc86477b)  

though it's not crashing without plugins

**Comment by domWalters** [comment]
I came across this bug today when porting my Neovim config to an Ubuntu 22.04 Virtual Machine.

# System Info

### Operating System
Ubuntu 22.04 in VirtualBox 7.0.10

### Neovim
v0.9.1 installed via Snap

### Terminals Used

- Alacritty (v0.12.2) with tmux (v3.2a-4)
- Gnome Terminal

# Plugins

I could reliably trigger it when certain plugins were installed at the same time:

- [noice.nvim](https://github.com/folke/noice.nvim) and [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) (unknown if parser choice affects this)
- [noice.nvim](https://github.com/folke/noice.nvim) and [markdown-preview.nvim](https://github.com/iamcco/markdown-preview.nvim)

After running Neovim for the first time and downloading packages using `packer.nvim`, I would close and reopen it, and consistently get the `to_scroll` assertion error with both plugin pairs listed above.

Removing `noice.nvim` prevented me from triggering the assertion failure.

Removing the two plugins on the right and keeping `noice.nvim` still resulted in the assertion failure, so I must have had at least one other plugin that had a problem.

I think it's worth mentioning that I use this same Neovim config on an Arch machine, and don't have this issue (same Neovim and Alacritty versions; I also ran a packer.nvim update and it still works; I have not tried installing packages from fresh).

# Possible Cause

I found the following two issues relating to `noice.nvim`:
- https://github.com/folke/noice.nvim/issues/53
  - Closed as not reproducable by issue owner or maintainer
- https://github.com/folke/noice.nvim/issues/372
  - Closed as not the plugin maintainers problem
  - Contains MCVE, although I haven't tried it

I think it's worth pointing out this bit from the `noice.nvim` README as the possible culprit:

> 🔥 Status
Noice is using the new experimental `vim.ui_attach` API, so issues are to be expected. It is highly recommended to use Neovim nightly, since a bunch of issues have already been fixed upstream. Check this tracking issue for a list of known issues.

**Comment by brandonpollack23** [comment]
This is happening to me right now on NVIM v0.11.0-dev-310+g9e436251de

**Comment by simone-lungarella** [comment]
Also happening to me whenever I try to open ~/.bashrc file, works well otherwise, apparently.

Nvim execute from bash of Pop!_OS having the following version:
```bash
nvim -v
NVIM v0.11.0-dev
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```
It seems to me that the problem is caused by `noice` plugin since the problem is solved if all noice configuration is removed.
